### ISSUE TITLE
Enable selenium-jupiter to provide EventFiringWebDriver

### ISSUE THREAD

**Issue by Bukama** [bonigarcia/selenium-jupiter#92]
_Feature request_

Often test automation devs don't only want to take screenshots at the end or record a session, but want to write a log during test execution to cover which buttons were pressed, which text was written into an input field and so on. Writing this over and over again, e.g. in PageObjects is bad practice, so a good way to write such a log is using an [`EventFiringWebDriver`](http://javadox.com/org.seleniumhq.selenium/selenium-support/docs/api/java/org/openqa/selenium/support/events/EventFiringWebDriver.html) which takes a `WebDriver` as an argument and registers itself to an [`WebDriverEventListener`](https://www.selenium.dev/selenium/docs/api/java/org/openqa/selenium/support/events/WebDriverEventListener.html). For a small example see [chapter 13 of Angie Jones selenium course at TAU](https://testautomationu.applitools.com/selenium-webdriver-tutorial-java/chapter13.html).

As I didn't want to wrap the `WebDriver` passed by selenium-jupiter in every test method, I tried to wrap another `ParameterResolve` around the selenium-jupiter one, because registering two is not possible at the moment (see [related JUnit issue](https://github.com/junit-team/junit5/issues/1604).) The way I tried was created by my [JUnit-Pioneer](https://junit-pioneer.org/) partner [Michael1993](https://github.com/Michael1993) (see [his post in the linked JUnit issue](https://github.com/junit-team/junit5/issues/1604#issuecomment-731755116)). 

But there are still some problems with that solution:
* The `@TestTemplate` provided by selenium-jupiter is not capable of `EventFiringWebDriver`, therefor the possibility to configure browsers in the `browsers.json` is not possible anymore
* It seems that some "after test work" gets broken as browsers often do not get closed

As of this and of course to provide this feature to all users I suggest to change the `@TestTemplate` of selenium-jupiter and always provide a `EventFiringWebDriver` which is already registered to an abstract class which implements the `WebDriverEventListener` interface (lets call it `SeleniumJupiterWebDriverEventListener` for the moment). With this all the nice features of selenium-jupiter are kept and every user, who wants to make use of an `WebDriverEventListener` can just add a class which extends the abstract `SeleniumJupiterWebDriverEventListener` to override the interface methods without the need to register a class on itself.

**Comment by bonigarcia** [comment]
I spent some time thinking and making some changes in the code about this request, but I am not completely sure about it. First, I don't like the idea to return always an `EventFiringWebDriver` instance in test templates. Instead, I would prefer that users select this type explicitly. Maybe something like this:

```
    @TestTemplate
    void templateTest(EventFiringWebDriver driver) {
    }
```
This can be done easily, but I don't see the advantage. I mean, it is basically the same to wrap the resulting `WebDriver` object with `EventFiringWebDriver`. Supposing the above code is possible, I suppose we still need some way to specify the `EventListener` you refer to (maybe using an annotation?).

In conclusion, I am not sure about this. Feel free to make a PR about it and we can discuss it in the code.
